internal/dag/gatewayapi_processor_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dag
+
+import (
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/projectcontour/contour/internal/fixture"
+	"github.com/stretchr/testify/assert"
+	gatewayapi_v1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+)
+
+func TestComputeHosts(t *testing.T) {
+	tests := map[string]struct {
+		route     *gatewayapi_v1alpha1.HTTPRoute
+		want      []string
+		wantError []error
+	}{
+		"single host": {
```

Comment:
Maybe add a test what has a hostname that is a single DNS label? Just to make sure `validation.IsDNS1123Subdomain` accepts that

Reply:
👍